spring-core-6/src/test/java/com/baeldung/httpinterface/BooksServiceMockitoTest.java

```diff
@@ -0,0 +1,102 @@
+    private Mono<ResponseEntity<Void>> monoResponseEntity;
+
+    @Mock
+    private ResponseEntity<Void> responseEntity;
+
+    @InjectMocks
+    private BooksClient booksClient;
+
+    @Test
+    void givenMockedWebClient_whenAllBooksAreRequested_thenTwoBooksAreReturned() {
+        when(webClient.method(HttpMethod.GET)).thenReturn(requestBodyUri);
+        when(requestBodyUri.uri(anyString(), anyMap())).thenReturn(requestBody);
+        when(requestBody.retrieve()).thenReturn(response);
+        when(response.bodyToMono(new ParameterizedTypeReference<List<Book>>(){}))
+          .thenReturn(Mono.just(List.of(
+            new Book(1,"Book_1", "Author_1", 1998),
+            new Book(2, "Book_2", "Author_2", 1999)
+          )));
+
+        BooksService booksService = booksClient.getBooksService();
+        List<Book> books = booksService.getBooks();
+        assertEquals(2, books.size());
+    }
+
+    @Test
+    void givenMockedWebClient_whenBookByIdIsRequested_thenCorrectBookIsReturned() {
+        when(webClient.method(HttpMethod.GET)).thenReturn(requestBodyUri);
+        when(requestBodyUri.uri(anyString(), anyMap())).thenReturn(requestBody);
+        when(requestBody.retrieve()).thenReturn(response);
+        when(response.bodyToMono(new ParameterizedTypeReference<Book>(){}))
+          .thenReturn(Mono.just(new Book(1,"Book_1", "Author_1", 1998)));
+
+        BooksService booksService = booksClient.getBooksService();
+        Book book = booksService.getBook(1);
+        assertEquals("Book_1", book.title());
+    }
+
+    @Test
+    void givenMockedWebClient_whenSaveNewBookIsRequested_thenCorrectBookIsReturned() {
+        when(webClient.method(HttpMethod.POST)).thenReturn(requestBodyUri);
+        when(requestBodyUri.uri(anyString(), anyMap())).thenReturn(requestBody);
+        when(requestBody.retrieve()).thenReturn(response);
+        when(response.bodyToMono(new ParameterizedTypeReference<Book>(){}))
+          .thenReturn(Mono.just(new Book(3, "Book_3", "Author_3", 2000)));
+
+        BooksService booksService = booksClient.getBooksService();
+        Book book = booksService.saveBook(new Book(3, "Book_3", "Author_3", 2000));
+        assertEquals("Book_3", book.title());
+    }
+
+    @Test
+    void givenMockedWebClient_whenDeleteBookIsRequested_thenCorrectCodeIsReturned() {
+        when(webClient.method(HttpMethod.DELETE)).thenReturn(requestBodyUri);
+        when(requestBodyUri.uri(anyString(), anyMap())).thenReturn(requestBody);
+        when(requestBody.retrieve()).thenReturn(response);
+        when(response.toBodilessEntity()).thenReturn(monoResponseEntity);
+        when(monoResponseEntity.block(any())).thenReturn(responseEntity);
+        when(responseEntity.getStatusCode()).thenReturn(HttpStatusCode.valueOf(200));
+
+        BooksService booksService = booksClient.getBooksService();
+        ResponseEntity<Void> response = booksService.deleteBook(3);
+        assertTrue(response.getStatusCode().is2xxSuccessful());
```

Comment:
Minor... this is a fair assertion, but if it fails all you get is "expected true, was false" when if you asserted the actual status code, you'd get "expected 200, was 401" or something more useful.